qCC/db_tree/ccDBRoot.cpp

```diff
@@ -1687,37 +1690,58 @@ void ccDBRoot::selectByTypeAndName()
 	scDlg.addType("Octree",            CC_TYPES::POINT_OCTREE);
 	scDlg.addType("Kd-tree",           CC_TYPES::POINT_KDTREE);
 	scDlg.addType("Viewport",          CC_TYPES::VIEWPORT_2D_OBJECT);
+	scDlg.addType("Custom Types",      CC_TYPES::CUSTOM_H_OBJECT);
 
 	if (!scDlg.exec())
 		return;
 
-	CC_CLASS_ENUM type = scDlg.getSelectedType();
-	QString name = scDlg.getSelectedName();
+	// for type checking
+	CC_CLASS_ENUM type = CC_TYPES::OBJECT; // all objects are matched by def
+	bool exclusive = false;
+
+	if (scDlg.getTypeIsUsed()) // we are using type checking
+	{
+		type = scDlg.getSelectedType();
+
+		//some types are exclusive, some are generic, and some can be both
+		//(e.g. Meshes)
+		//
+		//For generic-only types the match type gets overridden and forced to
+		//false because exclusive match makes no sense!
+		switch (type)
+		{
+		case CC_TYPES::HIERARCHY_OBJECT: //returned if no type is selected (i.e. all objects are selected!)
+		case CC_TYPES::PRIMITIVE:
+		case CC_TYPES::SENSOR:
+		case CC_TYPES::IMAGE:
+			exclusive = false;
+			break;
+		default:
+			exclusive = scDlg.getStrictMatchState();
+			break;
+		}
+	}
+
 
-	//some types are exclusive, some are generic, and some can be both
-	//(e.g. Meshes)
-	//
-	//For generic-only types the match type gets overridden and forced to
-	//false because exclusive match makes no sense!
-	bool exclusive;
-	switch (type)
+	// for name matching - def values
+	bool regex = false;
+	QString name = "";
```

Comment:
ok changed to QString name;

trying to understand why I wrote that is cause in these lines

```cpp
	// for name matching - def values
	bool regex = false;
	QString name; // an empty string by default
```
I am defining what will be the default values for two variables.. Thus I found clearer to state that _name_ is an empty string (not just a default constructed one...) and that _regex_ is false. 

Its like saying "I am not leaving the default values undefined - or def constructed -, but I am defining it as a length zero string... if you need you can write there something else if you need to change the default value".. that was my idea , probably:-)

thanks for the help btw